services/venice-controller/src/test/java/com/linkedin/venice/controller/TestVeniceParentHelixAdmin.java

```diff
@@ -2835,4 +2836,65 @@ public void testGetFinalReturnStatus() {
     finalStatus = VeniceParentHelixAdmin.getFinalReturnStatus(statuses, childRegions, 0, new StringBuilder());
     assertEquals(finalStatus, ExecutionStatus.DVC_INGESTION_ERROR_OTHER);
   }
+
+  @Test
+  public void testFilterStoresForCompaction() {
+    VeniceParentHelixAdmin mockAdmin = mock(VeniceParentHelixAdmin.class);
+    ArrayList<StoreInfo> storeInfoList = new ArrayList<>();
+    ArrayList<StoreInfo> compactionReadyStores = new ArrayList<>();
+
+    // Mock StoreInfo instances
+    StoreInfo store1 = mock(StoreInfo.class);
+    StoreInfo store2 = mock(StoreInfo.class);
+    StoreInfo store3 = mock(StoreInfo.class);
+
+    // Mock HybridStoreConfig for the first two StoreInfo instances
+    HybridStoreConfig hybridStoreConfig1 = mock(HybridStoreConfig.class);
+    HybridStoreConfig hybridStoreConfig2 = mock(HybridStoreConfig.class);
+    when(store1.getHybridStoreConfig()).thenReturn(hybridStoreConfig1);
+    when(store2.getHybridStoreConfig()).thenReturn(hybridStoreConfig2);
+    when(store3.getHybridStoreConfig()).thenReturn(null);
+
+    // Mock version numbers with random numbers
+    int store1VersionNumber = 1;
+    int store2VersionNumber = 2;
+    int store3VersionNumber = 3;
+    when(store1.getCurrentVersion()).thenReturn(store1VersionNumber);
+    when(store2.getCurrentVersion()).thenReturn(store2VersionNumber);
+    when(store3.getCurrentVersion()).thenReturn(store3VersionNumber);
+
+    // Mock Version instances
+    Version version1 = mock(Version.class);
+    Version version2 = mock(Version.class);
+    Version version3 = mock(Version.class);
+
+    // Return Version mocks when getVersion() is called
+    when(store1.getVersion(anyInt())).thenReturn(Optional.ofNullable(version1));
+    when(store2.getVersion(anyInt())).thenReturn(Optional.ofNullable(version2));
+    when(store3.getVersion(anyInt())).thenReturn(Optional.ofNullable(version3));
+
+    // Set createTime for Version mocks
+    long currentTime = System.currentTimeMillis();
+    long millisecondsPerHour = 60 * 60 * 1000;
+    when(version1.getCreatedTime()).thenReturn(currentTime - (25 * millisecondsPerHour)); // 25 hours ago
+    when(version2.getCreatedTime()).thenReturn(currentTime - (50 * millisecondsPerHour)); // 50 hours ago
+    when(version3.getCreatedTime()).thenReturn(currentTime - (23 * millisecondsPerHour)); // 23 hours ago
```

Comment:
This store is short-circuiting the check as it is a non-hybrid store.
Can you please add one more scenario to include a hybrid store that has had a version within the last 24 hours.

Reply:
version3 has a version 23 hours ago, which is <24 hours.

Reply:
True. However, version3 store is non-hybrid and hence the `shouldCompact` will already return false since its non-hybrid and in essence you are not validating the < 24 hour constraint explicitly.